Sources/AST/Expression/ExternalCall.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ExternalCall.swift
+//  AST
+//
+//  Created by Catalin Craciun on 10/11/2018.
+//
+import Source
+import Lexer
+
+/// A call to an external function
+public struct ExternalCall: ASTNode {
+  public var configurationParameters: [FunctionArgument]
+  public var functionCall: BinaryExpression
+  public var returnsOptional: Bool // used for representing call?
```

Comment:
Nit: the call does not return optional, `call?` is syntactic sugar for what is essentially

```swift
let result: Type?
do {
  result = call bla()
} catch {
  result = nil
}
```

Maybe something like `failsGracefullyOptional`